src/main/java/io/cdap/plugin/Run.java

```diff
@@ -59,24 +55,20 @@ public Run(RunConfig config) {
   public void configurePipeline(PipelineConfigurer pipelineConfigurer) throws IllegalArgumentException {
     StageConfigurer stageConfigurer = pipelineConfigurer.getStageConfigurer();
     Schema inputSchema = stageConfigurer.getInputSchema();
-    Preconditions.checkArgument(inputSchema != null, "Input Schema must be a known constant.");
-    config.validateBinaryExecutableType();
-    config.validateInputFields(inputSchema);
-    config.validateOutputFieldType();
-    if (!Strings.isNullOrEmpty(config.schema)) {
-      config.verifyOutputFieldTypeInSchema();
-    }
+    FailureCollector failureCollector = stageConfigurer.getFailureCollector();
+    config.validate(failureCollector, inputSchema);
```

Comment:
you forgot to check failure collector

Reply:
I not added failure collector check  based on the discussion with Vinisha
https://github.com/data-integrations/topn/pull/12#discussion_r323842824